pysqldb3/geopackage.py

```diff
@@ -201,69 +199,62 @@ def shp_to_gpkg(self, shp_name, gpkg_tbl = None, print_cmd=False):
         # check if the gpkg already exists. if so, we add another layer to the gpkg
         if not os.path.isfile(os.path.join(self.path, self.gpkg_name)):
 
-            self.cmd = WRITE_SHP_CMD_GPKG.format(gpkg_name = self.gpkg_name,
+            cmd = WRITE_SHP_CMD_GPKG.format(gpkg_name = self.gpkg_name,
+                                                  full_path = self.path,
                                                 shp_name = shp_name,
                                                 _update = '',
-                                                gpkg_tbl = gpkg_tbl,
-                                                export_path=self.path)
+                                                gpkg_tbl = gpkg_tbl)
         
         else:
 
-            self.cmd = WRITE_SHP_CMD_GPKG.format(gpkg_name = self.gpkg_name,
+            cmd = WRITE_SHP_CMD_GPKG.format(gpkg_name = self.gpkg_name,
+                                                  full_path = self.path,
                                                 shp_name = shp_name,
                                                 _update = '-update',
-                                                gpkg_tbl = gpkg_tbl,
-                                                export_path=self.path)
-
-        if print_cmd:
-            print(print_cmd_string([self.dbo.password], self.cmd))
+                                                gpkg_tbl = gpkg_tbl)
 
         try:
-            ogr_response = subprocess.check_output(shlex.split(self.cmd.replace('\n', ' ')), stderr=subprocess.STDOUT)
+            ogr_response = subprocess.check_output(shlex.split(cmd.replace('\n', ' ')), stderr=subprocess.STDOUT)
             print(ogr_response)
         except subprocess.CalledProcessError as e:
             print("Ogr2ogr Output:\n", e.output)
             print('Ogr2ogr command failed. The Geopackage/feature class was not written.')
-            raise subprocess.CalledProcessError(cmd=print_cmd_string([self.dbo.password], self.cmd), returncode=1)
-
         return
 
-    def gpkg_to_shp(self, print_cmd=False):
-        
+    def gpkg_to_shp(self, gpkg_tbl = None, export_path = None, cmd = None):
         """
         Converts a Geopackage to a Shapefile.
         The output Shapefile name will match the name of the geopackage table to be copied.
 
-        :param gpkg_name: file name for geopackage input (should end in .gpkg)
-
-        :param export_path: The folder directory to place the shapefiles output.
+        :param gpkg_name (str): file name for geopackage input (should end in .gpkg)
+        :param gpkg_tbl (str): OPTIONAL specific table within the geopackage to convert to a Shapefile
```

Comment:
could this be a list if you only want a subset of tables?

Reply:
(writing this comment for reference) I think what we agreed on was that if they weren't uploading all the tables in a geopackage, they would write a loop themselves to load the individual tables